bundles/org.eclipse.jface/src/org/eclipse/jface/resource/FileImageDescriptor.java

```diff
@@ -295,4 +306,13 @@ String getFilePath(String name, boolean logIOException) {
 			return null;
 		}
 	}
+
+	@Override
+	public <T> T getAdapter(Class<T> adapter) {
+		if (adapter == ImageFileNameProvider.class) {
+			// Support testing ImageFileNameProvider characteristics, see #396
+			return adapter.cast(new ImageProvider());
```

Comment:
Could an image provide maybe only instantiated once?

Reply:
Well, it could, but that's not how the FileImageDescriptor class is currently implemented. You should notice that it always generates a fresh new ImageProvider when it's required. And not also that the sole purpose is (currently) for testing. I would suggest that we should create a separate issue if we want to have only a single ImageFileNameProvider instance within the ImageProvider, because that would be a specific optimization approach.

Reply:
A corresponding issue has been created, see ##398.